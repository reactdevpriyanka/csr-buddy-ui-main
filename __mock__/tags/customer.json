[
  {
    "name": "HEARING_IMPAIRED",
    "value": true
  }
]
