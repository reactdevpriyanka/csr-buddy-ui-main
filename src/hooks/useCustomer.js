import axios from 'axios';
import useSWR from 'swr';
import { useRouter } from 'next/router';
import { SNACKVARIANTS } from '@components/SnackMessage/SnackMessage';
import { useSnackbar } from 'notistack';

export default function useCustomer(customerId) {
  const router = useRouter();

  const { id } = router.query;

  const queryId = customerId ? customerId : id;

  const { data, error, mutate } = useSWR(
    queryId ? `/api/v1/customer/${queryId}` : null,
    async (url) => axios.get(url).then(({ data }) => data),
    {
      revalidateIfStale: false,
      revalidateOnFocus: false,
      revalidateOnReconnect: false,
      shouldRetryOnError: true,
    },
  );

  const { enqueueSnackbar } = useSnackbar();

  const updateCustomer = async (customerId, customerData, onSuccess) =>
    axios
      .patch(`/api/v1/customer/${customerId}`, customerData)
      .then(() => onSuccess())
      .catch(() => {
        enqueueSnackbar({
          messageHeader: 'Error',
          variant: SNACKVARIANTS.ERROR,
          messageSubheader: 'Failed to Update Customer Details',
        });
      });

  return {
    data,
    error,
    updateCustomer,
    mutate,
  };
}
