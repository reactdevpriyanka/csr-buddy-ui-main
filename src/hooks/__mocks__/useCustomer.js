import customer from '__mock__/customers/nessa';

const mutate = jest.fn();

const mockUseCustomer = jest.fn(() => {
  return {
    mutate,
    error: null,
    data: customer,
    isValidating: false,
    updateCustomer: (id, data) =>
      new Promise((resolve) => {
        return resolve({ data: {} });
      }),
  };
});

mockUseCustomer.mutate = mutate;

export default mockUseCustomer;
