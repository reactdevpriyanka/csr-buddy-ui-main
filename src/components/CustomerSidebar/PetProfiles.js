import { useCallback } from 'react';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import Progress from '@material-ui/core/CircularProgress';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { Card, CardMedia, CardContent, CardActionArea } from '@mui/material';
import { Icon } from '@material-ui/core';
import AddCircleOutlineSharpIcon from '@material-ui/icons/AddCircleOutlineSharp';
import _ from 'lodash';
import { FeatureFlag } from '@/features';
import useAthena from '@/hooks/useAthena';
import useCustomer from '@/hooks/useCustomer';
import Link from '@mui/material/Link';
import PetAvatar from './PetProfile/PetAvatar';
import PetProfile from './PetProfile';

const useStyles = makeStyles((theme) => ({
  root: {
    padding: `${theme.spacing(1)} 0 0`,
  },
  header: {
    display: 'flex',
    flexFlow: 'row nowrap',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
  },
  heading: {
    ...theme.fonts.h2,
    marginBottom: theme.spacing(1.2),
    padding: '4.5px 0',
    color: theme.palette.blue.dark,
  },
  name: {
    ...theme.fonts.h4,
    fontSize: theme.fonts.size.md,
    color: theme.palette.blue.dark,
  },
  gridContainer: {
    marginBottom: `-${theme.spacing(4.5)}`,
    display: 'grid',
    rowGap: theme.spacing(0.5),
    padding: '5px 0 6px',
  },
  imgContainer: {
    padding: '12px',
  },
  img: {
    fontSize: 'medium',
    color: '#031657',
    transform: 'rotate(90deg)',
  },
  editPetLink: {
    background: 'transparent',
    border: 0,
    color: theme.palette.primary.alternate,
    fontSize: theme.fonts.size.sm,
    fontWeight: 700,
    cursor: 'pointer',
    lineHeight: theme.utils.fromPx(16.41),
    marginTop: theme.utils.fromPx(8),
    padding: 0,
    textTransform: 'uppercase',
  },
}));

const PetProfiles = ({ children }) => {
  const classes = useStyles();
  const { getLang } = useAthena();
  const router = useRouter();

  const { data, error } = useCustomer();

  const onCreatePet = useCallback(
    (event) => {
      event.preventDefault();
      router.push(
        {
          pathname: router.pathname,
          query: {
            ...router.query,
            interactionPanel: 'addPetProfile',
          },
        },
        undefined,
        { shallow: true },
      );
    },
    [router],
  );

  const onEditClick = useCallback(
    (event) => {
      event.preventDefault();
      router.push(
        {
          pathname: router.pathname,
          query: { ...router.query, interactionPanel: 'editPetProfile' },
        },
        undefined,
        { shallow: true },
      );
    },
    [router],
  );

  if (error) {
    return null;
  }

  if (!data) {
    return <Progress data-testid="loader" />;
  }

  const { pets = [] } = data;
  const addPetLabel = getLang('addPetlabel', { fallback: 'ADD A PET' });

  return (
    <section className={classes.root} data-testid="pet-profile">
      <div className={classes.header}>
        <Typography className={classes.heading} variant="h2">
          {getLang('petProfileslabel', { fallback: 'Pet Profiles' })}
        </Typography>
        {pets.length > 0 && (
          <FeatureFlag flag="feature.petProfile.editPetEnabled">
            <Link
              data-dd-action-name="Edit Pet Profile"
              onClick={onEditClick}
              variant="text"
              color="#1C49C2"
              component="span"
              data-testid="pet-profile:edit"
              className={classes.editPetLink}
              underline="hover"
            >
              {getLang('petProfilesEditlabel', { fallback: 'Edit' })}
            </Link>
          </FeatureFlag>
        )}
      </div>
      {pets.length > 0 && (
        <div className={classes.gridContainer}>
          {_.sortBy(pets, 'status').map((pet) => (
            <PetProfile id={pet.id} key={pet.id} petData={pet} />
          ))}
        </div>
      )}
      <FeatureFlag flag="feature.explorer.addPetEnabled">
        <div className={classes.gridContainer} data-testid="add-pet-container">
          <Card
            sx={{
              display: 'flex',
              height: 65,
              marginTop: 8.5,
              marginBottom: 4.5,
              borderRadius: 3,
            }}
          >
            <CardMedia sx={{ width: 80 }}>
              <PetAvatar
                data-testid="pet-profile:avatar"
                src="https://dotcms-ui.chewy.com/images/pet-profile/avatars/avatar_dog_xx_generic.png"
              />
            </CardMedia>
            <CardActionArea
              aria-label={addPetLabel}
              data-testid="pet-profile:add-pet-button"
              onClick={onCreatePet}
            >
              <CardContent>
                <Typography className={classes.name} component="div" variant="h6">
                  <Icon className={classes.imgContainer}>
                    <AddCircleOutlineSharpIcon className={classes.img} />
                  </Icon>
                  {addPetLabel}
                </Typography>
              </CardContent>
            </CardActionArea>
          </Card>
        </div>
      </FeatureFlag>
    </section>
  );
};

PetProfiles.propTypes = {
  children: PropTypes.node,
};

export default PetProfiles;
