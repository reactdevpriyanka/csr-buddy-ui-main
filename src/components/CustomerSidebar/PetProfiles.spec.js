import { renderWrap } from '@utils';
import useCustomer from '@/hooks/useCustomer';
import customer from '__mock__/customers/nessa';
import customerNoPets from '__mock__/customers/nessa+nopets';
import PetProfiles from './PetProfiles';

jest.mock('@/hooks/useCustomer');

describe('<PetProfiles />', () => {
  const render = renderWrap(PetProfiles);

  useCustomer.mockReturnValue({
    data: customer,
    error: null,
  });

  test('it should render profiles for customer pets', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when customer has no pets', () => {
    beforeAll(() => {
      useCustomer.mockReturnValue({
        data: customerNoPets,
        error: null,
      });
    });

    test('it should render no pets', () => {
      const { getByTestId } = render();
      const petAvatar = getByTestId('pet-profile:avatar');
      expect(petAvatar).toBeInTheDocument();
      expect(petAvatar.querySelector('img')).toHaveAttribute(
        'src',
        'https://dotcms-ui.chewy.com/images/pet-profile/avatars/avatar_dog_xx_generic.png',
      );
    });
  });

  describe('when error occurs', () => {
    beforeAll(() => {
      useCustomer.mockReturnValue({
        data: null,
        error: 'Some kind of fuzz',
      });
    });

    test('it should load nothing', () => {
      const { container } = render();
      expect(container.childNodes).toHaveLength(0);
    });
  });

  describe('when loading', () => {
    beforeAll(() => {
      useCustomer.mockReturnValue({
        data: null,
        error: null,
      });
    });

    test('it should render progress indicator', () => {
      const { getByTestId } = render();
      expect(getByTestId('loader')).toBeInTheDocument();
    });
  });
});
