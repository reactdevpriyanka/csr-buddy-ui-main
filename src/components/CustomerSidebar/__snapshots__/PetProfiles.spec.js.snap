// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PetProfiles /> it should render profiles for customer pets 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="makeStyles-root-1"
        data-testid="pet-profile"
      >
        <div
          class="makeStyles-header-2"
        >
          <h2
            class="MuiTypography-root makeStyles-heading-3 MuiTypography-h2"
          >
            Pet Profiles
          </h2>
          <span
            class="MuiTypography-root MuiTypography-text MuiLink-root MuiLink-underlineHover makeStyles-editPetLink-8 css-1ifudzl-MuiTypography-root-MuiLink-root"
            data-dd-action-name="Edit Pet Profile"
            data-testid="pet-profile:edit"
          >
            Edit
          </span>
        </div>
        <div
          class="makeStyles-gridContainer-5"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-12 css-bhp9pd-MuiPaper-root-MuiCard-root"
            data-testid="pet-profile:card"
          >
            <div
              class="MuiCardMedia-root css-16sznc5-MuiCardMedia-root"
            >
              <div
                class="MuiAvatar-root WithStyles(ForwardRef(Avatar))-root-23 WithStyles(ForwardRef(Avatar))-root-24 MuiAvatar-square"
                data-testid="pet-profile:avatar"
                gender="MALE"
              >
                <img
                  alt="pet-profile:avatar"
                  class="MuiAvatar-img"
                  src="https://dotcms-ui.chewy.com/images/pet-profile/avatars/default_dog.png"
                />
              </div>
            </div>
            <div
              class="MuiBox-root css-5hp7ux"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                data-dd-action-name="Pet Profile"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <div
                    class="makeStyles-mainPanel-18"
                  >
                    <div
                      class="makeStyles-topPanel-19"
                    >
                      <div
                        class="makeStyles-nameAge-21"
                      >
                        <div
                          aria-label="Booker"
                          class="makeStyles-toolTip-22"
                          data-mui-internal-clone-element="true"
                        >
                          <span>
                            <span>
                              <h6
                                class="MuiTypography-root makeStyles-name-14 MuiTypography-h6"
                                data-dd-action-name="Pet Profile"
                                data-testid="pet-profile:name"
                              >
                                Booker
                              </h6>
                              <span
                                class="makeStyles-confirmation-26"
                              >
                                <span
                                  style="display: flex; flex-flow: row nowrap; align-items: center;"
                                >
                                   
                                  Copied
                                </span>
                              </span>
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiTypography-root makeStyles-age-15 MuiTypography-h6"
                          data-testid="pet-profile:birthday"
                        >
                          <div
                            class="makeStyles-root-27"
                            data-testid="pet-profile:age"
                          >
                            <span
                              class="makeStyles-value-28"
                            >
                              3
                            </span>
                            YRS
                          </div>
                        </div>
                      </div>
                      <div
                        aria-label="Dog, Chihuahua"
                        class="makeStyles-toolTip-22"
                        data-mui-internal-clone-element="true"
                      >
                        <div
                          class="MuiTypography-root makeStyles-breedAndType-16 MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
                        >
                          <a
                            class="makeStyles-root-29"
                            data-testid="pet-profile:breedType"
                          >
                            Dog
                          </a>
                          , 
                          <a
                            class="makeStyles-root-30"
                            data-testid="pet-profile:breed"
                          >
                            Chihuahua
                          </a>
                        </div>
                      </div>
                    </div>
                    <div
                      class="makeStyles-bottomPanel-20"
                    />
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-gridContainer-5"
          data-testid="add-pet-container"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-qz1ock-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardMedia-root css-ketfxa-MuiCardMedia-root"
            >
              <div
                class="MuiAvatar-root WithStyles(ForwardRef(Avatar))-root-23 WithStyles(ForwardRef(Avatar))-root-31 MuiAvatar-circular"
                data-testid="pet-profile:avatar"
              >
                <img
                  class="MuiAvatar-img"
                  src="https://dotcms-ui.chewy.com/images/pet-profile/avatars/avatar_dog_xx_generic.png"
                />
              </div>
            </div>
            <button
              aria-label="ADD A PET"
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              data-testid="pet-profile:add-pet-button"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiTypography-root makeStyles-name-4 MuiTypography-h6"
                >
                  <span
                    aria-hidden="true"
                    class="material-icons MuiIcon-root makeStyles-imgContainer-6"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-img-7"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                  </span>
                  ADD A PET
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="makeStyles-root-1"
      data-testid="pet-profile"
    >
      <div
        class="makeStyles-header-2"
      >
        <h2
          class="MuiTypography-root makeStyles-heading-3 MuiTypography-h2"
        >
          Pet Profiles
        </h2>
        <span
          class="MuiTypography-root MuiTypography-text MuiLink-root MuiLink-underlineHover makeStyles-editPetLink-8 css-1ifudzl-MuiTypography-root-MuiLink-root"
          data-dd-action-name="Edit Pet Profile"
          data-testid="pet-profile:edit"
        >
          Edit
        </span>
      </div>
      <div
        class="makeStyles-gridContainer-5"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-12 css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="pet-profile:card"
        >
          <div
            class="MuiCardMedia-root css-16sznc5-MuiCardMedia-root"
          >
            <div
              class="MuiAvatar-root WithStyles(ForwardRef(Avatar))-root-23 WithStyles(ForwardRef(Avatar))-root-24 MuiAvatar-square"
              data-testid="pet-profile:avatar"
              gender="MALE"
            >
              <img
                alt="pet-profile:avatar"
                class="MuiAvatar-img"
                src="https://dotcms-ui.chewy.com/images/pet-profile/avatars/default_dog.png"
              />
            </div>
          </div>
          <div
            class="MuiBox-root css-5hp7ux"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              data-dd-action-name="Pet Profile"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="makeStyles-mainPanel-18"
                >
                  <div
                    class="makeStyles-topPanel-19"
                  >
                    <div
                      class="makeStyles-nameAge-21"
                    >
                      <div
                        aria-label="Booker"
                        class="makeStyles-toolTip-22"
                        data-mui-internal-clone-element="true"
                      >
                        <span>
                          <span>
                            <h6
                              class="MuiTypography-root makeStyles-name-14 MuiTypography-h6"
                              data-dd-action-name="Pet Profile"
                              data-testid="pet-profile:name"
                            >
                              Booker
                            </h6>
                            <span
                              class="makeStyles-confirmation-26"
                            >
                              <span
                                style="display: flex; flex-flow: row nowrap; align-items: center;"
                              >
                                 
                                Copied
                              </span>
                            </span>
                          </span>
                        </span>
                      </div>
                      <div
                        class="MuiTypography-root makeStyles-age-15 MuiTypography-h6"
                        data-testid="pet-profile:birthday"
                      >
                        <div
                          class="makeStyles-root-27"
                          data-testid="pet-profile:age"
                        >
                          <span
                            class="makeStyles-value-28"
                          >
                            3
                          </span>
                          YRS
                        </div>
                      </div>
                    </div>
                    <div
                      aria-label="Dog, Chihuahua"
                      class="makeStyles-toolTip-22"
                      data-mui-internal-clone-element="true"
                    >
                      <div
                        class="MuiTypography-root makeStyles-breedAndType-16 MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
                      >
                        <a
                          class="makeStyles-root-29"
                          data-testid="pet-profile:breedType"
                        >
                          Dog
                        </a>
                        , 
                        <a
                          class="makeStyles-root-30"
                          data-testid="pet-profile:breed"
                        >
                          Chihuahua
                        </a>
                      </div>
                    </div>
                  </div>
                  <div
                    class="makeStyles-bottomPanel-20"
                  />
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-gridContainer-5"
        data-testid="add-pet-container"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-qz1ock-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardMedia-root css-ketfxa-MuiCardMedia-root"
          >
            <div
              class="MuiAvatar-root WithStyles(ForwardRef(Avatar))-root-23 WithStyles(ForwardRef(Avatar))-root-31 MuiAvatar-circular"
              data-testid="pet-profile:avatar"
            >
              <img
                class="MuiAvatar-img"
                src="https://dotcms-ui.chewy.com/images/pet-profile/avatars/avatar_dog_xx_generic.png"
              />
            </div>
          </div>
          <button
            aria-label="ADD A PET"
            class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            data-testid="pet-profile:add-pet-button"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root makeStyles-name-4 MuiTypography-h6"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root makeStyles-imgContainer-6"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-img-7"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
                ADD A PET
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
